### PR TITLE
feat: consul registry support get services from multiple datacenters

### PR DESCRIPTION
**Pull request by woniu317**

### CODE REVIEW

File: contrib/registry/consul/client.go [hunk 1]
```diff
@@ -33,9 +41,14 @@ type Client struct {
 	serviceChecks api.AgentServiceChecks
 }
 
+type ClientConfig struct {
+	DCMode Datacenter
+}
+
 // NewClient creates consul client
-func NewClient(cli *api.Client) *Client {
+func NewClient(cli *api.Client, cfg *ClientConfig) *Client {
```
Comment (shenqidebaozi) [inline contrib/registry/consul/client.go:49]: This method should be marked as obsolete and need not be provided for external use

File: contrib/registry/consul/registry.go [hunk 2]
```diff
@@ -27,6 +27,13 @@ func WithHealthCheck(enable bool) Option {
 	}
 }
 
+// WithDCMode with registry datacenter option
+func WithDCMode(mode Datacenter) Option {
```
Comment (shenqidebaozi) [inline contrib/registry/consul/registry.go:31]: WithDatacenterMode

File: contrib/registry/consul/client.go [hunk 3]
```diff
@@ -33,15 +41,28 @@ type Client struct {
 	serviceChecks api.AgentServiceChecks
 }
 
+type ClientOption func(c *Client)
+
+// WithDataCenter datacenter names.
+func WithDataCenter(dc Datacenter) ClientOption {
```
Comment (shenqidebaozi) [inline contrib/registry/consul/client.go:47]: `WithDatacenter` would it be better
```
dcs, err := c.cli.Catalog().Datacenters()
```

**Review by shenqidebaozi — APPROVED** [review]
LGTM

**Review by tonybase — APPROVED** [review]

**Pull request merged**